### ISSUE TITLE
CA1812 false positive when type is only used in custom attribute

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn-analyzers#7787]
When a given internal type is only used in a custom attribute (eg. BDN's ConfigAttribibute), then CA1812 is reported for that type.

To reproduce, compile the following code fragment with **CA1812** enabled:
```c#
using System;

[Config(typeof(MultiFrameworkConfig))]
internal static class Program
{
    private static void Main(string[] args)
    {
        var list = Array.Empty<MultiFrameworkConfig>();
        Console.WriteLine(list.Length);
    }
}

[AttributeUsage(AttributeTargets.Class)]
public sealed class ConfigAttribute : Attribute
{
    public Type Type { get; }

    public ConfigAttribute(Type type)
    {
        Type = type;
    }
}

internal sealed class MultiFrameworkConfig
{
}
```

### Expected result

**CA1812** is not reported (for the **MultiFrameworkConfig** class).

### Actual result

```txt
Program.cs(24,23,24,43): error CA1812: 'MultiFrameworkConfig' is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make it 'static' (Module in Visual Basic). (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1812)
```
